### PR TITLE
Added support for translations to fetch from server for repeat group …

### PR DESCRIPTION
**Pull request by devanshhooda**
## Summary
This PR addresses the changes required for issue: [2795](https://github.com/dimagi/commcare-android/issues/2795)

## Product Description
We are now fetching the custom translations from the server for the elements of repeat groups.

## Safety Assurance

- [ ] I have confidence that this PR will not introduce a regression for the reasons below
- [x] Do we need to enhance manual QA test coverage ? If yes, "QA Note" label is set correctly

SS from commcarehq:
Screenshot 2024-09-11 at 4 25 36 PM

App SS:
Screenshot_20240911_162358
Screenshot_20240911_162419

**Release Note**: Adds support for custom translations for "Add" button in repeat group

### CODE REVIEW

File: app/src/org/commcare/activities/FormEntryActivityUIController.java [hunk 1]
```diff
@@ -498,8 +498,14 @@ private void createRepeatDialog() {
 
         // Assign title and text strings based on the current state
         String backText = Localization.get("repeat.dialog.go.back");
-        String addAnotherText = repeatCaptionPrompt.getRepeatText(FormEntryActivity.mFormController.getLastRepeatCount() > 0 ? "add" : "add-empty");
+
+        // Checking the repetitions here
```
Comment (shubham1g5) [inline app/src/org/commcare/activities/FormEntryActivityUIController.java:502]: It's a bad practice to introduce self explanatory comments in the code and comments should only be used when needed due to a counter-intuitive or complex piece of code.

**Review by shubham1g5 — CHANGES_REQUESTED** [review]
Changes looks right, added a suggestion to remove comments.

File: app/src/org/commcare/activities/FormEntryActivityUIController.java (hunk 1 above)
Comment (devanshhooda) [reply app/src/org/commcare/activities/FormEntryActivityUIController.java]: Done

File: app/src/org/commcare/activities/FormEntryActivityUIController.java [hunk 2]
```diff
@@ -493,20 +494,23 @@ private void createRepeatDialog() {
...
         } else {
             skipText = Localization.get("repeat.dialog.exit");
         }
-        if (FormEntryActivity.mFormController.getLastRepeatCount() > 0) {
-            title = Localization.get("repeat.dialog.add.another", FormEntryActivity.mFormController.getLastGroupText());
-        } else {
-            title = Localization.get("repeat.dialog.add.new", FormEntryActivity.mFormController.getLastGroupText());
-        }
+
+        title = addAnotherText;
```
Comment (shubham1g5) [inline app/src/org/commcare/activities/FormEntryActivityUIController.java:513]: nit: can combine assignment with declaration here.

**Review by shubham1g5 — DISMISSED** [review]

File: app/src/org/commcare/activities/FormEntryActivityUIController.java (hunk 2 above)
Comment (devanshhooda) [reply app/src/org/commcare/activities/FormEntryActivityUIController.java]: We don't even need that variable hence removed it now.

**Comment by devanshhooda** [comment]
@damagatachi retest this please

**Comment by devanshhooda** [comment]
@damagatachi retest this please

**Review by shubham1g5 — APPROVED** [review]

**Pull request merged**